language/tools/move-cli/src/lib.rs

```diff
@@ -0,0 +1,293 @@
+        if resource_path.is_dir() {
+            bail!("Bad resource path {:?}. Needed file, found directory")
+        }
+        match resource_path.file_name() {
+            None => bail!(
+                "Bad resource path {:?}; last component must be a file",
+                resource_path
+            ),
+            Some(name) => Ok({
+                let id = match parser::parse_type_tag(&name.to_string_lossy())? {
+                    TypeTag::Struct(s) => s,
+                    t => bail!("Expected to parse struct tag, but got {}", t),
+                };
+                match Self::get_bytes(resource_path)? {
+                    Some(resource_data) => Some(
+                        MoveValueAnnotator::new_no_stdlib(self)
+                            .view_resource(&id, &resource_data)?,
+                    ),
+                    None => None,
+                }
+            }),
+        }
+    }
+
+    pub fn view_module(&self, module_path: &Path) -> Result<Option<String>> {
+        type Loc = u64;
+        if module_path.is_dir() {
+            bail!("Bad module path {:?}. Needed file, found directory")
+        }
+
+        Ok(match Self::get_bytes(module_path)? {
+            Some(module_bytes) => {
+                // TODO: find or create source map and pass it to disassembler
+                let d: Disassembler<Loc> = Disassembler::from_module(
+                    CompiledModule::deserialize(&module_bytes)
+                        .map_err(|e| anyhow!("Failure deserializing module: {:?}", e))?,
+                    0,
+                )?;
+                Some(d.disassemble()?)
+            }
+            None => None,
+        })
+    }
+
+    /// Delete resource stored on disk at the path `addr`/`tag`
+    pub fn delete_resource(&self, addr: AccountAddress, tag: StructTag) -> Result<()> {
+        let path = self.get_resource_path(addr, tag);
+        fs::remove_file(path)?;
+
+        // delete addr directory if this address is now empty
+        let addr_path = self.get_addr_path(&addr);
+        if addr_path.read_dir()?.next().is_none() {
+            fs::remove_dir(addr_path)?
+        }
+        Ok(())
+    }
+
+    /// Save `resource` on disk under the path `addr`/`tag`
+    pub fn save_resource(
+        &self,
+        addr: AccountAddress,
+        tag: StructTag,
+        layout: MoveTypeLayout,
+        resource: Value,
+    ) -> Result<()> {
+        let mut path = self.get_addr_path(&addr);
+        if !path.exists() {
+            fs::create_dir(path.clone())?
+        }
+        path.push(StructID(tag).to_string());
+        let lcs = resource
+            .simple_serialize(&layout)
+            .ok_or_else(|| anyhow!("Failed to serialize resource"))?;
+        let mut f = fs::File::create(path)?;
+        Ok(f.write_all(&lcs)?)
+    }
+
+    /// Save `module` on disk under the path `module.address()`/`module.name()`
```

Comment:
I love how the "directory" structure of modules and resources are now literally directories